beluga_system_tests/test/test_system_new.cpp

```diff
@@ -144,37 +144,17 @@ auto particle_filter_test(
                    ranges::views::take_exactly(params.max_particles) |            //
                    ranges::to<beluga::TupleVector>;
 
-  // Don't update the filter unless we've moved far enough from the latest pose where an update was performed.
-  // It's an approximation of the recommendations Probabilistic Robotics \cite thrun2005probabilistic
-  // Chapter 4.2.4 based in Nav2 AMCL's implementation .
-  // TODO(nahuel): This should be part of the library (in some form).
-  auto far_enough_to_update = [params, latest_pose = std::optional<Sophus::SE2d>{}](auto pose) mutable -> bool {
-    if (!latest_pose) {
-      latest_pose = pose;
-      return true;
-    }
-
-    const auto delta = latest_pose->inverse() * pose;
-    const bool delta_is_above_threshold =                           //
-        std::abs(delta.translation().x()) > params.update_min_d ||  //
-        std::abs(delta.translation().y()) > params.update_min_d ||  //
-        std::abs(delta.so2().log()) > params.update_min_a;
-
-    if (delta_is_above_threshold) {
-      latest_pose = pose;
-    }
-
-    return delta_is_above_threshold;
-  };
+  auto should_update = beluga::policies::on_motion(params.update_min_d, params.update_min_a);
 
-  // TODO(nahuel): This should be part of the library (in some form).
-  auto should_resample = [params, count = 0UL]() mutable -> bool {
-    return ++count % params.resample_interval_count == 0;
-  };
+  using Particles = decltype(particles);
+  beluga::any_policy<Particles> should_resample = beluga::policies::every_n(params.resample_interval_count);
```

Comment:
@nahueespinosa meta^N: I hate that C++ template argument deduction isn't powerful enough and kinda forces us to use `decltype` here.

Reply:
+100